lib/platform/github/index.ts

```diff
@@ -1719,3 +1725,64 @@ export async function getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]> {
   }
   return alerts;
 }
+
+async function pushFiles(
+  { branchName, message }: CommitFilesConfig,
+  { parentCommitSha, commitSha }: CommitResult
+): Promise<CommitSha | null> {
+  try {
+    await pushCommitToRenovateRef(commitSha, branchName);
+    const treeItems = await listCommitTree(commitSha);
+
+    const treeRes = await githubApi.postJson<{ sha: string }>(
+      `/repos/${config.repository}/git/trees`,
+      { body: { tree: treeItems } }
+    );
+    const treeSha = treeRes.body.sha;
+
+    const commitRes = await githubApi.postJson<{ sha: string }>(
+      `/repos/${config.repository}/git/commits`,
+      { body: { message, tree: treeSha, parents: [parentCommitSha] } }
+    );
+    const remoteCommitSha = commitRes.body.sha;
+
+    try {
+      await githubApi.postJson(`/repos/${config.repository}/git/refs`, {
+        body: { ref: `refs/heads/${branchName}`, sha: remoteCommitSha },
+      });
+    } catch (e) {
+      // istanbul ignore if
+      if (e.err?.code !== 'ERR_NON_2XX_3XX_RESPONSE') {
+        throw e;
+      }
+      await githubApi.patchJson(
+        `/repos/${config.repository}/git/refs/heads/${branchName}`,
+        { body: { sha: remoteCommitSha, force: true } }
+      );
+    }
```

Comment:
Can you document this POST then PATCH? Is it to first try to create a branch and if that fails then update it? Can't we already know whether a branch exists or not?

Reply:
Yes, I'm unsure how to better handle this. One more option was to perform HEAD request, but this way we perform 2 guaranteed requests and still get 404 logged to the DEBUG level.

Reply:
Fixed and checked it works for real-world repo (https://github.com/renovate-testing/test-11537-github-commit-api/pulls)